.github/actions/make_init/action.yml

```diff
@@ -0,0 +1,77 @@
+name: "Initialize Virtual Env"
+runs:
+  using: composite
+  steps:
+    - name: Restore nvm, pre-commit cache, & node modules from cache
+      id: cache-nvm-node-modules
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.nvm
+          ~/.cache
+          frontend/node_modules
```

Comment:
It is not recommended to cache `node_modules` directory on GH. See: https://github.com/actions/cache/pull/537

Reply:
I'd be okay with bending the rules here a bit (as long as we include a comment explaining why) if it's significantly faster to cache `node_modules` since

* we don't use `npm` as our package manager, so `npm ci` not working is irrelevant to us
* we don't need to support multiple `node` versions

That being said, if we're not seeing significant performance gains from caching `node_modules`, then I'd also be for just not doing so to follow the recommendations

Reply:
Resolved via implementation of github action setup-node 👍🏼